csrc/runtime/executor_dispatch.h

```diff
@@ -16,14 +16,20 @@ namespace nvfuser {
 // ExprEvalExecutor
 class ExecutorDispatch {
  public:
-  // Iterates through executors in priority order creating the first executor
-  // that returns true when checking their "supported" method
+  // If `scheduler_type` is `SchedulerType::None`, this function Iterates
+  // through executors in priority order creating the first executor that
+  // returns true when checking their "supported" method. Otherwise, create the
+  // executor according to `scheduler_type`, which is faster.
+  //
+  // The slow path (i.e. `SchedulerType::None`) is only used by
+  // MultiDeviceExecutor at this moment.
```

Comment:
How does the `MultiDeviceExecutor` infer the scheduler / segmentation? 
Or does it only use `SchedulerType::Communication`?

Reply:
MultiDeviceExecutor doesn't infer the scheduler type and therefore calls this method with `SchedulerType::None`. 

Let me know how you'd like the code comment above to clarify this.

Reply:
For my understanding of the stack:
`MultiDeviceExecutor::runWithInputs` invokes `HostIrEvaluator`. The `handle(PostOnStream* post_ir)` will call `makeExecutor`. I am missing where the scheduler is being inferred? Can you point me to it?